pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1036,11 +1084,11 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) error
 				return nil
 			}
 
-			if added, err := p.addToActiveQ(logger, pInfo); !added {
+			if added, err := p.moveToActiveQ(logger, pInfo, BackoffComplete); !added {
 				return err
 			}
+			// Need to call an additional delete to remove usPodInfo.Pod which might differ from pInfo.Pod removed in moveToActiveQ.
```

Comment:
why would it differ? Isn't only the name of the pod what matters?

Reply:
Right, misread the code above